### PR TITLE
feat(record): _RecordArrayView + parent-identity sweep grouping

### PR DESCRIPTION
**Pull request by jhuggins**
Makes Pattern A (`f(p=design)`) and Pattern B (`f(**design.select_all())`) produce identical sweep outputs, via a scaled-back form of the view contract discussed in #138.

## What changes

- **New `_RecordArrayView` class** (`probpipe/core/_record_array.py`). A single-field `RecordArray` subclass that aliases one column from a parent RA and carries the parent reference as shared-identity metadata. Minimal surface — conversion (`__array__` / `__jax_array__`), introspection (`.shape` / `.dtype` / `.ndim`), indexing, iteration, `len()`, plus `.parent` / `.field`. No operator forwarding, no array-reduction shims — users `jnp.asarray(view)` before arithmetic, consistent with the existing `NumericRecord` single-field-shim policy.

- **`RecordArray.view(field)`** accessor — opt-in construction of a view. `ra["field"]` itself still returns the raw column.

- **`Design.select_all()`** now returns views (`self.view(f)` for each field).

- **WF sweep grouping** (`probpipe/core/node.py`). `_broadcast` groups array-valued sweep args by parent identity: views sharing a parent form one group and iterate in lockstep (zip); distinct parents (independent `RecordArray`s, independent `DistributionArray`s, views from different parents) each form their own group and product across the sweep. Within-group shape mismatches raise.

## What this buys

```python
ff = FullFactorialDesign(r=[1.5, 1.8, 2.0], K=[60.0, 80.0])

# Pattern A — single Record-typed arg
@workflow_function
def fit_a(p: NumericRecord):
    return p["r"] * p["K"]
out_a = fit_a(p=ff)                 # NumericRecordArray, batch=(6,)

# Pattern B — splat via select_all
@workflow_function
def fit_b(r, K):
    return r * K
out_b = fit_b(**ff.select_all())    # NumericRecordArray, batch=(6,)

# Identical output — parent-identity grouping zips the two views.
assert out_a.batch_shape == out_b.batch_shape == (6,)
```

Independent array args still cartesian-product (no parent shared):

```python
ra_a = NumericRecordArray.stack([NumericRecord(a=float(i)) for i in range(3)])
ra_b = NumericRecordArray.stack([NumericRecord(b=float(j * 10)) for j in range(2)])

@workflow_function
def f(a, b): return a + b

out = f(a=ra_a.view("a"), b=ra_b.view("b"))     # batch=(3, 2) — cartesian
```

## Intentional non-change

`ra["field"]` still returns the raw column `jnp.ndarray`. The fuller version of the contract (views as default) would need ~31 test migrations and careful design around operator forwarding / `__getattr__` magic / multi-inheritance with `NumericRecordArray`. Tracked in #138 as an open design question — ship this narrower version first; flip the contract in a follow-up if/when we settle the open questions there.

## Tests

30 new tests in `tests/test_record_array_view.py`:

- **Construction + parent identity** (7 tests): `ra.view(f)` builds a view; alias semantics; unknown field raises; raw-column behaviour unchanged; mixed numeric/object fields.
- **Attribute forwarding** (10 tests): `.shape` / `.dtype` / `.ndim`; `jnp.asarray(view)` / `np.asarray(view)`; indexing / slicing / iteration / `len`; arithmetic requires explicit conversion (documents the policy).
- **Sweep grouping** (4 tests): two views same parent zip; two views different parents product; view + plain RA product; three sibling views collapse to single axis.
- **Pattern A ≡ Pattern B parity** (3 tests): numeric-only FullFactorialDesign; mixed numeric/categorical; `select_all()` returns views.
- **Raw columns stay independent** (1 test): `f(x=ra["x"], y=ra["y"])` runs body once with JAX broadcasting — no zip sweep, since raw columns carry no parent-identity signal.
- **Mixed composition** (2 tests): view + DistributionArray (product); view + scalar Distribution (nested regime).

Full suite: **2169 passed, 7 skipped, 0 failed** (was 2139; +30 new).

## Stack

Builds on PR #137 (Design hierarchy). Once #137 rebases onto main, this rebases cleanly on top of it.

## Related issues

- #138 — tracks the open design question: should `ra["field"]` return a view by default? Shipped this PR deliberately without the contract change; if we decide to flip, that's a separate follow-up PR.

### CODE REVIEW

**Comment by jhuggins** [comment]
Replaced by #141 — the old PR was auto-closed when its base branch `dev/design-hierarchy` was deleted after merging #140. Same branch head (`dev/record-array-views`), rebased onto main.
